huaweicloud/services/iam/resource_huaweicloud_identity_service_agency.go

```diff
@@ -0,0 +1,539 @@
+	detachPath := client.Endpoint + detachHttpUrl
+	detachPath = strings.ReplaceAll(detachPath, "{policy_id}", policyID)
+	detachOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		JSONBody: map[string]interface{}{
+			"agency_id": agencyID,
+		},
+	}
+	_, err := client.Request("POST", detachPath, &detachOpt)
+	if err != nil {
+		return fmt.Errorf("error detaching IAM service agency to policy(%s): %s", policyID, err)
+	}
+	return nil
+}
+
+func createTags(createTagsClient *golangsdk.ServiceClient, tags map[string]interface{}, resourceType, id string) error {
+	if len(tags) > 0 {
+		createTagsHttpUrl := "v5/{resource_type}/{resource_id}/tags/create"
+		createTagsPath := createTagsClient.Endpoint + createTagsHttpUrl
+		createTagsPath = strings.ReplaceAll(createTagsPath, "{resource_type}", resourceType)
+		createTagsPath = strings.ReplaceAll(createTagsPath, "{resource_id}", id)
+		createTagsOpt := golangsdk.RequestOpts{
+			KeepResponseBody: true,
+			JSONBody: map[string]interface{}{
+				"tags": expandResourceTags(tags),
+			},
+		}
+
+		_, err := createTagsClient.Request("POST", createTagsPath, &createTagsOpt)
+		if err != nil {
+			return fmt.Errorf("error creating tags: %s", err)
+		}
+	}
+	return nil
+}
+
+func expandResourceTags(tagmap map[string]interface{}) []map[string]interface{} {
+	var taglist []map[string]interface{}
+	for k, v := range tagmap {
+		tag := map[string]interface{}{
+			"tag_key":   k,
+			"tag_value": v,
+		}
+		taglist = append(taglist, tag)
+	}
+	return taglist
+}
+
+func deleteTags(deleteTagsClient *golangsdk.ServiceClient, tags map[string]interface{}, resourceType, id string) error {
+	if len(tags) > 0 {
+		deleteTagsHttpUrl := "v5/{resource_type}/{resource_id}/tags/delete"
+		deleteTagsPath := deleteTagsClient.Endpoint + deleteTagsHttpUrl
+		deleteTagsPath = strings.ReplaceAll(deleteTagsPath, "{resource_type}", resourceType)
+		deleteTagsPath = strings.ReplaceAll(deleteTagsPath, "{resource_id}", id)
+		deleteTagsOpt := golangsdk.RequestOpts{
+			KeepResponseBody: true,
+			JSONBody:         expandTagsKeyToStringList(tags),
+		}
+
+		_, err := deleteTagsClient.Request("DELETE", deleteTagsPath, &deleteTagsOpt)
+		if err != nil {
+			return fmt.Errorf("error deleting tags: %s", err)
+		}
+	}
+	return nil
+}
+
+func expandTagsKeyToStringList(tagmap map[string]interface{}) []string {
+	var tagKeyList []string
+	for k := range tagmap {
+		tagKeyList = append(tagKeyList, k)
+	}
+	return tagKeyList
+}
+
+func resourceIAMServiceAgencyDelete(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+	client, err := cfg.NewServiceClient("iamv5", region)
+	if err != nil {
+		return diag.Errorf("error creating IAM v5 client: %s", err)
+	}
+
+	// before deleting service agency, require to detach policies first
+	// get policy ID which will be detached
+	policyIDs, err := getPolicyIDsByNames(client, d.Get("policy_names").(*schema.Set).List())
```

Comment:
can we get the attached policies with `/v5/agencies/{agency_id}/attached-policies`?